affinity/cpp-20/d0796r3.md

```diff
@@ -348,31 +376,32 @@ execution::execution_context execContext(systemResource[0]);
 auto &execResource = execContext.resource();
 
 // systemResource[0] should be equal to execResource
-
-for (const execution::execution_resource &res : execResource) {
-    std::cout << res.name()  << "\n";
-}
 ```
-*Listing 5: Example of constructing an execution context from an execution resource*
+*Listing 5: Example of constructing an `execution_context` from an `execution_resource`*
 
-When creating an `execution_context` from a given `execution_resource`, the executors and allocators associated with it are bound to that `execution_resource`. For example, when creating an `execution_resource` from a CPU socket resource, all executors associated with the given socket will spawn execution agents with affinity to the socket partition of the system *(Listing 6)*.
+When creating an `execution_context` from a given `execution_resource`, the executors associated with it are bound to that `execution_resource` and any `execution_resource`s encapsulated by it. For example, when creating an `execution_resource` from a CPU socket resource, all executors associated with the given socket will spawn execution agents with affinity to the socket partition of the system *(Listing 6)*.
 
 ```cpp
 auto systemResource = std::this_system::discover_topology();
-// FindASocketResource is a user-defined function that finds a 
-// resource that is a CPU socket in the given resource list
-auto& socket = findASocketResource(systemResource);
-execution_contextC{socket} // Associated with the socket
-auto executor = eC.executor(); // By transitivity, associated with the socket too
-auto socketAllocator = eC.allocator(); // Retrieve an allocator to the closest memory node
-std::vector<int, decltype(socketAllocator)> v1(100, socketAllocator);
-std::generate(par.on(executor), std::begin(v1), std::end(v1), std::rand);
+
+/* find_socket_resource is a user-defined function that finds a resource that is a CPU socket in the given resource list */
+auto socket = find_socket_resource(systemResource);
+
+/* Create an execution_context and executor associated with the CPU socket */
+execution_context context{socket}
+auto executor = context.executor();
+
+/* Create an allocator from the memory resource associate with the GPU socket */
+pmr::polymorphic_allocator<int> alloc{socket.memory_resource()};
+
+pmr::vector<int> vec(100, alloc);
+std::generate(par.on(executor), std::begin(vec), std::end(vec), std::rand);
```

Comment:
`std::rand` is not thread safe, so it's not so good to use for the example here.

Reply:
That's a good point, I've replaced `std::rand` with a generic `genFunc` for now, but we should try to improve the motivational examples in the future.

Reply:
The reason this is `pmr::vector` is because we're constructing the vector using a `pmr::polynmorphic_allocator`. It could also be a regular `std::vector` with the `polymorphic_allocator` specified explicitly.